### ISSUE TITLE
Username is not defined

### ISSUE THREAD

**Issue by jdahdah** [AndreasFaust/gatsby-source-custom-api#36]
There seems to be an issue with the new auth object. It's supposed to be optional, but if I look at the [line](https://github.com/AndreasFaust/gatsby-source-custom-api/pull/28/files#diff-864cfd1dc7f1900a1561837fba2390a4756443cc348ec1bf28b0156c2196becaR21) in the [PR](https://github.com/AndreasFaust/gatsby-source-custom-api/pull/28), there is no check whether that object is actually provided or not.

This is resulting in the following error and breaking the plugin before it manages to download anything.

```
TypeError: Cannot read property 'username' of undefined
    at createImageNodes
(/Users/home/Development/test/node_modules/gatsby-source-custom-api/utils/loadImages.js:21:35)
    at /Users/home/Development/test/node_modules/gatsby-source-custom-api/utils/loadImages.js:96:14
    at runNextTicks (node:internal/process/task_queues:61:5)
    at processImmediate (node:internal/timers:437:9)
    at async Promise.all (index 5)
    at Object.exports.sourceNodes
(/Users/home/Development/test/node_modules/gatsby-source-custom-api/gatsby-node.js:47:14)
    at runAPI (/Users/home/Development/test/n
```

**Comment by cdennington** [comment]
Trying to install via NPM, Its saying 2.3.2 is the latest and this issue is still present:
<img width="373" alt="image" src="https://user-images.githubusercontent.com/1114197/158004839-71702ee0-8153-4fd7-9d27-4b507965b484.png">
If this was fixed in 2.3.3, how can I upgrade?

**Comment by jdahdah** [comment]
Indeed it looks like v2.3.3 was never published to [npm](https://www.npmjs.com/package/gatsby-source-custom-api).
